File: DatingApp-Api/Helpers/UserParams.cs
namespace DatingApp_Api.Helpers
{
    public class UserParams : PageInationParams
    {

        public string CurrentUserName { get; set; }
        public string Gender { get; set; }

        public int MinAge { get; set; } = 18;
        public int MaxAge { get; set; } = 99;

        public string OrderBy { get; set; } = "lastActive";






    }
}

File: DatingApp-client/src/app/_models/userParams.ts
import { User } from "./User";

export class UserParams {
    gender: string;
    minAge: number = 18;
    maxAge: number = 99;
    pageNumber: number = 1;
    pageSize: number = 2;
    orderBy: string = "lastActivity";

    constructor(user: User) {
        this.gender = user.gender === 'male' ? 'female' : 'male';
    }
}